tests/subsys/openthread/radio_test.c

```diff
@@ -0,0 +1,818 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+
+#include <zephyr.h>
+#include <errno.h>
+#include <stdio.h>
+
+#include <net/ieee802154_radio.h>
+#include <net/net_pkt.h>
+
+#include <openthread/platform/radio.h>
+#include <platform-zephyr.h>
+
+/**
+ * @brief Tests for the radio.c - OpenThread radio api
+ * @defgroup openthread_tests radio
+ * @ingroup all_tests
+ * @{
+ */
+
+#define ACK_PKT_LENGTH 3
+#define FRAME_TYPE_MASK 0x07
+#define FRAME_TYPE_ACK 0x02
+
+K_SEM_DEFINE(ot_sem, 0, 1);
+
+energy_scan_done_cb_t scan_done_cb;
+
+/**
+ * Fake pointer as it should not be accessed by the code.
+ * Should not be null to be sure it was properly passed.
+ */
+otInstance *ot = (otInstance *)0xAAAA;
+
+/* forward declarations */
+static int scan_mock(struct device *dev, u16_t duration,
+		     energy_scan_done_cb_t done_cb);
+static enum ieee802154_hw_caps get_capabilities(struct device *dev);
+static int cca_mock(struct device *dev);
+static int set_channel_mock(struct device *dev, u16_t channel);
+static int filter_mock(struct device *dev, bool set,
+		       enum ieee802154_filter_type type,
+		       const struct ieee802154_filter *filter);
+static int set_txpower_mock(struct device *dev, s16_t dbm);
+static int tx_mock(struct device *dev, enum ieee802154_tx_mode mode,
+		   struct net_pkt *pkt, struct net_buf *frag);
+static int start_mock(struct device *dev);
+static int stop_mock(struct device *dev);
+static int configure_mock(struct device *dev, enum ieee802154_config_type type,
+			  const struct ieee802154_config *config);
+
+/* mocks */
+static struct ieee802154_radio_api rapi = { .get_capabilities =
```

Comment:
You could move all of the initializations one line below, with a single tab indentation to avoid line breaks.